lib/sisimai/bite/email/exim.rb

```diff
@@ -0,0 +1,522 @@
+
+            elsif e['command'] == 'MAIL'
+              # MAIL | Connected to 192.0.2.135 but sender was rejected.
+              e['reason'] = 'onhold'
+
+            else
+              # Verify each regular expression of session errors
+              ReFailure.each_key do |r|
+                # Check each regular expression
+                next unless e['diagnosis'] =~ ReFailure[r]
+                e['reason'] = r.to_s
+                break
+              end
+
+              unless e['reason']
+                # The reason "expired"
+                e['reason'] = 'expired' if e['diagnosis'] =~ ReDelayed
+              end
+            end
+          end
+
+          # Prefer the value of smtp reply code in Diagnostic-Code:
+          # See eg/maildir-as-a-sample/new/exim-20.eml
+          #   Action: failed
+          #   Final-Recipient: rfc822;userx@test.ex
+          #   Status: 5.0.0
+          #   Remote-MTA: dns; 127.0.0.1
+          #   Diagnostic-Code: smtp; 450 TEMPERROR: retry timeout exceeded
+          # The value of "Status:" indicates permanent error but the value
+          # of SMTP reply code in Diagnostic-Code: field is "TEMPERROR"!!!!
+          sv = Sisimai::SMTP::Status.find(e['diagnosis'])
+          rv = Sisimai::SMTP::Reply.find(e['diagnosis'])
+          s1 = 0  # First character of Status as integer
+          r1 = 0  # First character of SMTP reply code as integer
+
+          # "Status:" field did not exist in the bounce message
+          if sv.empty?
+            # Check SMTP reply code
+            if rv.size > 0
+              # Generate pseudo DSN code from SMTP reply code
+              r1 = rv[0, 1].to_i
+              if r1 == 4
+                # Get the internal DSN(temporary error)
+                sv = Sisimai::SMTP::Status.code(e['reason'], true)
+
+              elsif r1 == 5
+                # Get the internal DSN(permanent error)
+                sv = Sisimai::SMTP::Status.code(e['reason'], false)
+              end
+            end
+          end
+
+          s1  = sv[0, 1].to_i if sv.size > 0
+          v1  = s1 + r1
+          v1 += e['status'][0, 1].to_i if e['status']
+
+          if v1 > 0
+            # Status or SMTP reply code exists
+            # Set pseudo DSN into the value of "status" accessor
+            e['status'] = sv if r1 > 0
+          else
+            # Neither Status nor SMTP reply code exist
+            if e['reason'] =~ /\A(?:expired|mailboxfull)/
+              # Set pseudo DSN (temporary error)
+              sv = Sisimai::SMTP::Status.code(e['reason'], true)
+
+            else
+              # Set pseudo DSN (permanent error)
+              sv = Sisimai::SMTP::Status.code(e['reason'], false)
+            end
+          end
+          e['status'] ||= sv
+          e.each_key { |a| e[a] ||= '' }
+        end
+
+        rfc822part = Sisimai::RFC5322.weedout(rfc822list)
+        return { 'ds' => dscontents, 'rfc822' => rfc822part }
+      end
+
```

Comment:
__[rubocop]__
* Layout/CommentIndentation: Incorrect indentation detected (column 14 instead of 19).

<!-- Comment by [SideCI](https://sideci.com) -->